app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  include SessionsHelper
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
+      session[:user_id] = user.id
+      redirect_to root_url
+    else
+      render 'new'
+    end
+  end
+
+  def destroy
+    log_out if logged_in?
+    redirect_to root_url
+  end
+
+  def log_out
+    session[:user_id] = nil
+  end
```

Comment:
`log_out` doesn't appear to be used outside of `destroy`. I'd suggest moving this code inline in `destroy` to minimize indirection.